omod/src/main/java/org/openmrs/module/spa/servlet/SpaServlet.java

```diff
@@ -17,13 +23,25 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import static org.openmrs.module.util.SingleSpaConstants.DEFAULT_SPA_BASE_URL;
+import static org.openmrs.module.util.SingleSpaConstants.GP_KEY_SPA_BASE_URL;
+
 public class SpaServlet extends HttpServlet {
 
 	private static final long serialVersionUID = 1L;
+	private static final Logger log = LoggerFactory.getLogger(SpaServlet.class);
 
 	@Override
 	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
 		RequestDispatcher dispatcher = req.getRequestDispatcher("/module/spa/master-single-page-application.form");
+		req.setAttribute("openmrsBaseUrlContext", new String(req.getContextPath()));
+		try {
+			req.setAttribute("spaBaseUrlContext", Context.getAdministrationService().getGlobalProperty(GP_KEY_SPA_BASE_URL,
+					DEFAULT_SPA_BASE_URL));
+		} catch (APIException e) {
```

Comment:
Isn't this an anti-pattern to just log and rethrow the same exception?

Reply:
How do I go about it? Intention is to return HTTP 500 error

Reply:
Since you are not processing the exception but just logging and rethrowing, why not leave it to the caller?

Reply:
That means getting rid of the try-catch block?

Reply:
Correct.

Reply:
I have effected the changes as agreed